### DISCUSSION TITLE
vscode extension relentlessly generating .metals folder, when it probably shouldn't

### DISCUSSION THREAD

**Discussion by dj-mc** [scalameta/metals#3158] [Q&A] [answered]
The vscode extension is generating a `.metals` folder with a `metals.lock.db` file, but it's doing this in projects that have nothing to do with Scala. e.g. a workspace with nothing but a .cxx source file. It sometimes does this after I click on the extension icon. **If I delete the folder it's regenerated instantly.**

Sometimes this happens and sometimes it doesn't. Restarting vscode is a temporary fix, but I would like to know what's happening and can I prevent it?

**Comment by tgodzik** [comment, marked as answer]
This is most likely caused by VS Code timing out and enabling the Metals extension by default. 

The condition that times out is https://github.com/scalameta/metals-vscode/blob/main/package.json#L53 and it's problem with VS Code, which I don't believe should activate extensions in case of timeout.

You should see the exact reason, why Metals was activated in `Output` view after choosing `log (extension host)`
